src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[3];
```

Comment:
The code assumes that the separator is at index 3 of the `fromFormat` array. This assumption might not hold true if the format is different, such as 'YYYY-MM-DD'. Consider revising the logic to dynamically determine the separator based on the actual date string or format pattern.